src/random.cpp

```diff
@@ -192,20 +192,38 @@ uint64_t GetRdSeed() noexcept
 #elif defined(__aarch64__) && defined(HWCAP2_RNG)
 
 bool g_rndr_supported = false;
+bool g_rndrrs_supported = false;
+
+bool VerifyRNDRRS()
+{
+    uint8_t ok;
+    uint64_t test;
+    int max_retries = 10;
+    do {
+        __asm__ volatile("mrs %0, s3_3_c2_c4_1; cset %w1, ne;"
+                         : "=r"(test), "=r"(ok)::"cc");
+        if (ok) return true;
+        __asm__ volatile("yield");
+    } while (--max_retries > 0);
+    return false;
+}
 
 void InitHardwareRand()
 {
     if (getauxval(AT_HWCAP2) & HWCAP2_RNG) {
         g_rndr_supported = true;
+        g_rndrrs_supported = VerifyRNDRRS();
     }
 }
 
 void ReportHardwareRand()
 {
     // This must be done in a separate function, as InitHardwareRand() may be indirectly called
     // from global constructors, before logging is initialized.
-    if (g_rndr_supported) {
+    if (g_rndr_supported && g_rndrrs_supported) {
         LogPrintf("Using RNDR and RNDRRS as additional entropy sources\n");
```

Comment:
It seems neither of these are available on Apple Silicon (`HWCAP2_RNG` isn't defined), so not much to test here.